### PR TITLE
SONAR-10008 Make issue search date histogram use server time zone

### PR DESCRIPTION
**Pull request by bartfastiel**

### CODE REVIEW

**Comment by bartfastiel** [comment]
We are waiting for @stas-vilchik 's commit before we merge. Probably a functional verification by a product manager would also be a good idea.

File: server/sonar-server/src/main/java/org/sonar/server/es/Facets.java [hunk 1]
```diff
@@ -128,14 +135,20 @@ private void processSubAggregations(HasAggregations aggregation) {
...
-        facet.put(value.getKeyAsString(), Math.round(((Sum) value.getAggregations().get(FACET_MODE_EFFORT)).getValue()));
+        facet.put(day, Math.round(((Sum) value.getAggregations().get(FACET_MODE_EFFORT)).getValue()));
       } else {
-        facet.put(value.getKeyAsString(), value.getDocCount());
+        facet.put(day, value.getDocCount());
       }
     }
   }
 
+  private String dateTimeToDate(String timestamp, TimeZone timeZone) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/es/Facets.java]: MAJOR Make "dateTimeToDate" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: sonar-plugin-api/src/main/java/org/sonar/api/utils/DateUtils.java [hunk 2]
```diff
@@ -54,6 +55,10 @@ public static String formatDate(Date d) {
     return d.toInstant().atZone(ZoneId.systemDefault()).toLocalDate().toString();
   }
 
+  public static String formatDate(Date d, TimeZone timeZone) {
```
Comment (henryju) [inline sonar-plugin-api/src/main/java/org/sonar/api/utils/DateUtils.java]: I'm not sure we should add more stuff in the plugin API. It's always a mess to maintain. What about keeping it private?

Comment (bartfastiel) [reply sonar-plugin-api/src/main/java/org/sonar/api/utils/DateUtils.java]: 👍

**Pull request merged**